cms/bundles/models.py

```diff
@@ -83,6 +86,34 @@ def previewable(self) -> Self:
     def annotate_release_date(self) -> "BundlesQuerySet":
         return self.annotate(release_date=models.F("release_date"))  # type: ignore[no-any-return]
 
+    def annotate_queryset_updated_at(self) -> "BundlesQuerySet":
+        # Annotate the objects' updated_at using the latest log entry's timestamp
+        log_model = log_registry.get_log_model_for_model(self.model)
+
+        # If the log model is not a subclass of ModelLogEntry, we don't know how
+        # to query the logs for the object, so skip the annotation.
+        if not log_model or not issubclass(log_model, ModelLogEntry):
+            return self
+
+        latest_log = (
+            log_model.objects.filter(
+                content_type=ContentType.objects.get_for_model(self.model, for_concrete_model=False),
+                object_id=Cast(models.OuterRef("pk"), models.CharField()),
+            )
+            .order_by("-timestamp", "-pk")
+            .values("timestamp")[:1]
+        )
+        return self.annotate(updated_at=models.Subquery(latest_log))  # type: ignore[no-any-return]
+
+    def annotate_status_label(self) -> "BundlesQuerySet":
```

Comment:
This is used for ordering by the status label on the index view, rather than the `BundleStatus` value, so that "Draft" (value: "DRAFT") is before "Ready to publish" (value: "APPROVED")